### ISSUE TITLE
Right column on documentation not fully visible

### ISSUE THREAD

**Issue by melix** [gradle/gradle#10159]
### Expected Behavior

The table of contents on the right column on any page should be fully readable or scrolling. 

### Current Behavior

Today the right column seems to be fixed and prevents from reading the full TOC.

![scrolling](https://user-images.githubusercontent.com/316357/62375086-c7ba0400-b53d-11e9-8ca6-bc89f882ffde.gif)

### Steps to Reproduce 

Run `./gradlew userguide -x userguideSinglePage` on `master`.
Then open `/subprojects/docs/build/docs/userguide/dependency_management.html`

Scroll down, see that the right column won't move.

### Your Environment

Firefox 68.0.1

**Comment by ohubaut** [comment]
So, I took a look at the CSS, and the issue is that, above a certain real estate on screen (width > 75 rem & height > 48 rem), the position of the div containing the Table of Content (`#toc`) is assigned a `position: fixed` attribute. 

Fixed, means, well, fixed 🙂, no scroll allowed.

There are several ways to solve the issue, but it is hard to properly address the issue given the lack of information at hand:

- I could not find any document describing the design system in place for the website, nor even a simple set of mockups for expected layouts at specific resolutions (or rather windows size);
- I could not find what were the requirements in term of legacy broker support. This could drastically reduce the available options;
- the CSS use a strange mix of BEM, CSS-by-ID and ad-hoc rules, making it very difficult to estimate the impact of a local change;
- I could not figure out why the problematic attribute was in.a media query that expected a certain window's height, and I fear I'm missing something here.

At this point, all I can propose, is to go with a solution that is grid-based (no support for IE then). A simple example of the solution can be seen [on this codepen](https://codepen.io/gizmogwai/pen/YzyoYpx).
You'll need to resize the browser to see it in action.
However, touching the current CSS might require a lot of caution given its actual state.

**Comment by eriwen** [comment]
I'm going to provide 2 solutions and I'll let you folks choose:

**Option 1: quick uglyish fix**
The technical reason the right column isn't scrollable is because the TOC container elements' `height` are all `auto`.

All you need to do to fix this is add the following to a pre-existing rule matching the TOC containers.

```css
.appendix #header #toc, .book #header #toc, .chapter #header #toc {
    overflow: auto;
    /* 350px = the height of the header + height of the footer */
    height: calc(100vh - 350px);
}
```

This allows the TOC container itself to scroll. 

![Screenshot 2020-06-01 09 33 12](https://user-images.githubusercontent.com/51534/83431156-be165c80-a3eb-11ea-93f9-7e405cb1fbc3.png)

**Option 2: fix information hierarchy**
You can apply the option 1 fix here, but this should make it unnecessary.

First of all, this `dependency_management` page is way too long and focuses on too many subtopics. The fact that we implemented TOC structure just for this page indicates this. Stepping back a little: if I, as a user, wanted to learn the basics of Gradle dependency management, I would want an overview and links to subtopics I'm interested in. If I had to guess, this is doc is nearly 50 pages if we printed it. This is a lot like lines of code in a method, it's not wrong per se, but I would be lying if it didn't look overwhelming and intimidating to me.

One of the negative side effects of how we've structured this page is that we've started introducing top-level titles way down in the content like this:

![Screenshot 2020-06-01 09 38 49](https://user-images.githubusercontent.com/51534/83431143-b951a880-a3eb-11ea-9515-16440c19550f.png)

This negatively affects search indexing and isn't styled because the system isn't expecting a top-level `h1` here. Try for your yourself by searching for "Working in a multi-repo environment". 

So specifically I personally recommend:
 * Break this chapter up into 5 or so, having the same structure but on separate pages. Link to them! 
 * Add those new pages to navigation. Search index will automatically pick em up soon.
 * Apply option 1 above if you like.

I don't think it's all that much more work and in my opinion provides a better user experience. HTH

**Comment by melix** [comment]
Just commenting on this one because I don't have the CSS skills for the rest:
> First of all, this dependency_management page is way too long and focuses on too many subtopics.

This page is an aggregator of all DM topics. The DM topics are _already_ split into different sections in the main userguide page, under "Working with dependencies", but there are two things which really annoy me:

1. as a user, I want to get _all the DM docs at once_, without the rest, which is why I like this aggregating page
2. the "Working with dependencies" section on the [userguide](https://docs.gradle.org/current/userguide/userguide.html) as well as all main sections in the left TOC don't have deep link anchors I can copy and paste to share for users. Which means I cannot point to "the DM docs". All I can do is say "look at the userguide and search for dependency management" _or_ point at _one section_ of the DM docs, which is usually not what I want.

In other words, DM is a book in itself, and I'd very much like to have an option to get it all :)

**Comment by eriwen** [comment]
I see. I also see now that this is unlisted in the navigation.

Users have 2 other ways to get "all the docs": the PDF and the [single-page userguide](https://docs.gradle.org/current/userguide/userguide_single.html). 

I'm not sure I fully understand "main sections in the left TOC don't have deep link anchors". You can find permalinks to any subheading/subsection by hovering over the header, but you probably already knew that. 

![Screenshot 2020-06-02 08 34 39](https://user-images.githubusercontent.com/51534/83539525-fe3d1400-a4ab-11ea-91a4-7c982715cc3f.png)

If you really want the DM docs to be a separate book, then I suggest you copy the logic used for the PDF and userguide_single which handles the TOC in a much better way.

**Comment by melix** [comment]
> I'm not sure I fully understand "main sections in the left TOC don't have deep link anchors

I'd like to have an hover anchor on "Working with dependencies" here:

![Selection_497](https://user-images.githubusercontent.com/316357/83540647-e5a71b00-a4f8-11ea-9634-1f5d2e815cdc.png)

But there's none.

**Comment by eriwen** [comment]
Aha, interesting. If I give you https://docs.gradle.org/current/userguide/userguide.html#managing-dependencies does that satisfy your needs here @melix?
